app/src/main/java/app/pandorapass/pandora/ui/pages/Passwords.kt

```diff
@@ -0,0 +1,340 @@
+                    )
+                }
+                item {
+                    OutlinedTextField(
+                        modifier = width,
+                        value = newUsername,
+                        onValueChange = { newUsername = it },
+                        label = { Text("Username") },
+                        singleLine = true
+                    )
+                }
+                item {
+                    OutlinedTextField(
+                        modifier = width,
+                        value = newPassword,
+                        onValueChange = { newPassword = it },
+                        label = { Text("Password") },
+                        visualTransformation =
+                            if (!showPassword) PasswordVisualTransformation()
+                            else VisualTransformation.None,
+                        trailingIcon = {
+                            IconButton(onClick = { showPassword = !showPassword }) {
+                                Icon(
+                                    imageVector = ImageVector.vectorResource(
+                                        if (showPassword) R.drawable.eye_slash_24_outlined
+                                        else R.drawable.eye_24_outlined
+                                    ),
+                                    contentDescription = null
+                                )
+                            }
+                        },
+                        singleLine = true
+                    )
+                }
+                items(urls) { url ->
+                    Text(url)
+                }
+                item {
+                    OutlinedTextField(
+                        modifier = width,
+                        value = newURL,
+                        onValueChange = { newURL = it },
+                        label = { Text("URL") },
+                        trailingIcon = {
+                            IconButton(onClick = { urls = urls + newURL; newURL = "" }) {
+                                Icon(
+                                    imageVector = ImageVector.vectorResource(R.drawable.plus_24_outlined),
+                                    contentDescription = null
+                                )
+                            }
+                        },
+                        singleLine = true
+                    )
+                }
+                item {
+                    OutlinedTextField(
+                        modifier = width,
+                        value = newNotes,
+                        onValueChange = { newNotes = it },
+                        label = { Text("Notes") },
+                        singleLine = true
+                    )
+                }
+                item {
+                    Button(
+                        modifier = width,
+                        enabled = (newURL.isNotBlank() && newUsername.isNotBlank() && newPassword.isNotBlank()),
```

Comment:
title, username, and password are the only required fields, so the url field can be empty.